lib/alerts/check_timeline_alert_manager.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+class CheckTimelineAlertManager
+  def initialize(course)
+    @course = course
+    create_alerts
+  end
+
+  def create_alerts
+    return unless @course.approved?
+    # if course doesn't have any training module, it will not create an alert
+    return if @course.training_module_ids.any?
+    return if alert_already_exists?(@course.id)
```

Comment:
There's no need to pass the id as an argument to this method, since the instance variable `@course` is automatically available to every instance method.

Reply:
@ragesoss I've updated everything mentioned. And Now it's not generating duplicate alert when there is already an existing alert. You can have a look and try it on your local